### DISCUSSION TITLE
Extension activation failed: "Response content-type is text/plain; charset=utf-8 (status=404)"

### DISCUSSION THREAD

**Discussion by tpu01yzx** [community/community#75280] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Information from VS Code:

> Version: 1.84.2
> Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
> Date: 2023-11-09T10:51:52.184Z
> Electron: 25.9.2
> ElectronBuildId: 24603566
> Chromium: 114.0.5735.289
> Node.js: 18.15.0
> V8: 11.4.183.29-electron.0
> OS: Windows_NT x64 10.0.19045

Extesion:

> GitHub.copilot v1.135.0

It can be logined locally, but not in Remote-SSH which produces error:

> Extension activation failed: "Response content-type is text/plain; charset=utf-8 (status=404)"

**Comment by davecheney** [comment]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by tpu01yzx** [reply]
It may be not the case since Remote-SSH machine and local machine are in the same local network. In fact, Remote-SSH machine is a docker container while local machine is a KVM-Qemu virtual machine. They are in the same physical machine. The physical machine is under my control and I did not install any software intercepting SSL traffic. 

Is there any suggestion for debuging the activation process so that I can check it manually from the shell in Remote-SSH machine ?

**Comment by tpu01yzx** [comment]
From https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot#diagnosing-network-issues , more info is shown as below.

> $ curl --verbose https://copilot-proxy.githubusercontent.com/_ping
> *   Trying 20.250.58.93:443...
> * Connected to copilot-proxy.githubusercontent.com (20.250.58.93) port 443 (#0)
> * ALPN, offering h2
> * ALPN, offering http/1.1
> *  CAfile: /etc/ssl/certs/ca-certificates.crt
> *  CApath: /etc/ssl/certs
> * TLSv1.0 (OUT), TLS header, Certificate Status (22):
> * TLSv1.3 (OUT), TLS handshake, Client hello (1):
> * TLSv1.2 (IN), TLS header, Certificate Status (22):
> * TLSv1.3 (IN), TLS handshake, Server hello (2):
> * TLSv1.2 (IN), TLS header, Finished (20):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Certificate (11):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, CERT verify (15):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Finished (20):
> * TLSv1.2 (OUT), TLS header, Finished (20):
> * TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.3 (OUT), TLS handshake, Finished (20):
> * SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
> * ALPN, server accepted to use h2
> * Server certificate:
> *  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=copilot-proxy.githubusercontent.com
> *  start date: Jul 19 00:00:00 2023 GMT
> *  expire date: Jul 19 23:59:59 2024 GMT
> *  subjectAltName: host "copilot-proxy.githubusercontent.com" matched cert's "copilot-proxy.githubusercontent.com"
> *  issuer: C=US; O=DigiCert Inc; CN=DigiCert Global G2 TLS RSA SHA256 2020 CA1
> *  SSL certificate verify ok.
> * Using HTTP2, server supports multiplexing
> * Connection state changed (HTTP/2 confirmed)
> * Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * Using Stream ID: 1 (easy handle 0x55da10ef9e90)
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> > GET /_ping HTTP/2
> > Host: copilot-proxy.githubusercontent.com
> > user-agent: curl/7.81.0
> > accept: */*
> > 
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * Connection state changed (MAX_CONCURRENT_STREAMS == 10)!
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> < HTTP/2 200 
> < content-security-policy: default-src 'none'; sandbox
> < content-type: application/json
> < strict-transport-security: max-age=31536000
> < content-length: 48
> < date: Wed, 15 Nov 2023 02:11:02 GMT
> < 
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> {"now":1700014262,"status":"ok","ns1":"200 OK"}
> * Connection #0 to host copilot-proxy.githubusercontent.com left intact

**Reply by davecheney** [reply]
I believe the problem is your request to api.github.com was rejected

**Reply by tpu01yzx** [reply, marked as answer]
> $ curl --verbose https://api.github.com                           
> *   Trying 140.82.112.4:443...
> * Connected to api.github.com (140.82.112.4) port 443 (#0)
> * ALPN, offering h2
> * ALPN, offering http/1.1
> *  CAfile: /etc/ssl/certs/ca-certificates.crt
> *  CApath: /etc/ssl/certs
> * TLSv1.0 (OUT), TLS header, Certificate Status (22):
> * TLSv1.3 (OUT), TLS handshake, Client hello (1):
> * TLSv1.2 (IN), TLS header, Certificate Status (22):
> * TLSv1.3 (IN), TLS handshake, Server hello (2):
> * TLSv1.2 (IN), TLS header, Finished (20):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Certificate (11):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, CERT verify (15):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Finished (20):
> * TLSv1.2 (OUT), TLS header, Finished (20):
> * TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.3 (OUT), TLS handshake, Finished (20):
> * SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
> * ALPN, server accepted to use h2
> * Server certificate:
> *  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
> *  start date: Feb 16 00:00:00 2023 GMT
> *  expire date: Mar 15 23:59:59 2024 GMT
> *  subjectAltName: host "api.github.com" matched cert's "*.github.com"
> *  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS Hybrid ECC SHA384 2020 CA1
> *  SSL certificate verify ok.
> * Using HTTP2, server supports multiplexing
> * Connection state changed (HTTP/2 confirmed)
> * Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * Using Stream ID: 1 (easy handle 0x557a163a3e90)
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> > GET / HTTP/2
> > Host: api.github.com
> > user-agent: curl/7.81.0
> > accept: */*
> > 
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
> * old SSL session ID is stale, removing
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> * TLSv1.2 (OUT), TLS header, Supplemental data (23):
> * TLSv1.2 (IN), TLS header, Supplemental data (23):
> < HTTP/2 301 
> < content-length: 0
> < location: https://github.com/
> < 
> * Connection #0 to host api.github.com left intact

Yes, you are right. I do not why in Remote-SSH machine, the IP of _api.github.com_ is the same as that of _github.com_.  Anyway, after adding the following two lines in _/etc/hosts_, everything works well now.

> 140.82.112.4 github.com
> 140.82.113.6 api.github.com

**Comment by akhilpatlolla** [comment]
```2024-10-28 15:00:06.314 [info] [certificates] Removed 11 expired certificates
2024-10-28 15:00:07.728 [error] [auth] auth: JsonParseError: Response content-type is text/html; charset=utf-8 (status=403)
    at EH.json (c:\Users\P3268834\.vscode\extensions\github.copilot-1.243.0\lib\src\networking.ts:145:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at authFromGitHubToken (c:\Users\P3268834\.vscode\extensions\github.copilot-1.243.0\lib\src\auth\copilotToken.ts:132:54)
    at auth (c:\Users\P3268834\.vscode\extensions\github.copilot-1.243.0\extension\src\auth.ts:40:25)
    at authShowWarnings (c:\Users\P3268834\.vscode\extensions\github.copilot-1.243.0\extension\src\auth.ts:131:25)
    at tae.getCopilotToken (c:\Users\P3268834\.vscode\extensions\github.copilot-1.243.0\extension\src\auth.ts:174:33)
    at attemptAuthentication (c:\Users\P3268834\.vscode\extensions\github.copilot-1.243.0\extension\src\auth.ts:58:9) {
  code: 'ContentType=text/html; charset=utf-8'
}
2024-10-28 15:00:07.728 [error] [auth] Response content-type is text/html; charset=utf-8 (status=403)
```



output for curl 

```
$ curl --verbose https://api.github.com
* Host api.github.com:443 was resolved.
* IPv6: (none)
* IPv4: 140.82.114.6
*   Trying 140.82.114.6:443...
* Connected to api.github.com (140.82.114.6) port 443
* schannel: disabled automatic use of client certificate
* schannel: next InitializeSecurityContext failed: CRYPT_E_NO_REVOCATION_CHECK (0x80092012) - The revocation function was unable to check revocation for the certificate.
* Closing connection
* schannel: shutting down SSL/TLS connection with api.github.com port 443
curl: (35) schannel: next InitializeSecurityContext failed: CRYPT_E_NO_REVOCATION_CHECK (0x80092012) - The revocation function was unable to check revocation for the certificate.```

**Reply by Intevel** [reply]
Currently having this
